### ISSUE TITLE
Cannot used with `struct`

### ISSUE THREAD

**Issue by honghaoz** [devxoul/Then#16]
Seems like I cannot use `Then` on struct type:

``` swift
struct SomeStruct {
    var name: String = ""
}

extension SomeStruct : Then {}

var aStruct = SomeStruct().then {
    $0.name = "123"
}
```

`$0.name = "123"` line throw an error: `...swift:43:10: Cannot assign to property: '$0' is a 'let' constant`

**Comment by devxoul** [comment]
@honghaoz, #18 fixes the issue and has been shipped with 1.0.0 release :smile:

**Comment by honghaoz** [comment]
great! thanks :+1:
